share/smack/lib/smack.c

```diff
@@ -2053,18 +2053,17 @@ void __SMACK_decls() {
   D("procedure {:inline 1} $$alloc(n: ref) returns (p: ref)\n"
     "modifies $Alloc, $CurrAddr;\n"
     "{\n"
-    "  p := $CurrAddr;\n"
-    "  havoc $CurrAddr;\n"
+    "  assume $sge.ref.bool(n, $0.ref);\n"
```

Comment:
Won’t the assumption of `n >= 0` cause programs that do `malloc(-1)` to verify trivially? Is this situation somehow averted by type conversions?